frontend/src/lib/components/CompareFilter/CompareFilter.tsx

```diff
@@ -1,30 +1,70 @@
 import { LemonCheckbox } from '@posthog/lemon-ui'
 import { useActions, useValues } from 'kea'
+import { RollingDateRangeFilter } from 'lib/components/DateFilter/RollingDateRangeFilter'
+import { uuid } from 'lib/utils'
+import { useState } from 'react'
+import { useRef } from 'react'
 import { insightLogic } from 'scenes/insights/insightLogic'
 import { insightVizDataLogic } from 'scenes/insights/insightVizDataLogic'
 
+import { getDefaultComparisonPeriodRelativeStartDate } from './compareFilterLogic'
+
 export function CompareFilter(): JSX.Element | null {
+    const key = useRef(uuid()).current
     const { insightProps, canEditInsight } = useValues(insightLogic)
 
-    const { compare, supportsCompare } = useValues(insightVizDataLogic(insightProps))
+    const { compare, comparison, supportsCompare, dateRange, interval } = useValues(insightVizDataLogic(insightProps))
     const { updateInsightFilter } = useActions(insightVizDataLogic(insightProps))
 
     const disabled: boolean = !canEditInsight || !supportsCompare
+    const defaultRelativePeriod = getDefaultComparisonPeriodRelativeStartDate(dateRange, interval)
+    const [comparisonRelativePeriodStart, setComparisonRelativePeriodStart] = useState(
+        comparison?.relative_period_start || defaultRelativePeriod
+    )
 
     // Hide compare filter control when disabled to avoid states where control is "disabled but checked"
     if (disabled) {
         return null
     }
 
     return (
-        <LemonCheckbox
-            onChange={(compare: boolean) => {
-                updateInsightFilter({ compare })
-            }}
-            checked={!!compare}
-            label={<span className="font-normal">Compare to previous period</span>}
-            bordered
-            size="small"
-        />
+        <>
+            <LemonCheckbox
+                onChange={(compare: boolean) => {
+                    if (!compare) {
+                        updateInsightFilter({ compare: undefined, comparison: undefined })
+                    } else {
+                        updateInsightFilter({
+                            compare,
+                            comparison: {
+                                relative_period_start: comparisonRelativePeriodStart,
+                            },
+                        })
+                    }
+                }}
+                checked={!!compare}
+                label={<span className="font-normal">Compare to the period starting</span>}
+                size="small"
+                className="ml-4"
+            />
+            <RollingDateRangeFilter
+                pageKey={key}
+                dateFrom={comparisonRelativePeriodStart}
```

Comment:
There's a problem here. When I update the insight range, the comparison period is not updated, because RollingDateRangeFilter stores the current values in the state. Here's a [video](https://github.com/PostHog/posthog/assets/6707908/a3810c53-1953-4d1a-87f4-23e0879a6f6d) of what I mean, "2 days" comparison period remains the same.

IIUC if I want to update the comparison period when the range updates, I need to
1. create compareFilterLogic
2. Connect it to the insightLogic; listen to the updateInsightFilter
3. On updateInsightFilter call a method that would update RollingDateRangeFilter state from the new comparison period

Is that correct? If so, I suggest I implement it in the follow up PR, WDYT? Maybe there's a simpler way to do this?